starknet/docs/its.md

```diff
@@ -0,0 +1,574 @@
+                                     generation) (env: STARKNET_PRIVATE_KEY)
+  --accountAddress <accountAddress>  Starknet account address (env:
+                                     STARKNET_ACCOUNT_ADDRESS)
+  -h, --help                         display help for command
+
+Note: This command deploys tokens locally via the InterchainTokenFactory.
+- If initialSupply > 0, tokens will be minted to the deployer and mintership transferred to the minter
+- If initialSupply = 0, the minter address will be set directly
+- For cross-chain deployment, use a separate command after local deployment
+```
+
+**Examples:**
+
+Deploy with initial supply:
+```bash
+npx ts-node starknet/its/deploy-token.ts --salt my-token-salt --name "My Token" --symbol "MTK" --decimals 18 --initialSupply 1000000 --minter 0x123... --privateKey 0x... --accountAddress 0x...
+```
+
+Deploy without initial supply (minter gets set directly):
+```bash
+npx ts-node starknet/its/deploy-token.ts --salt my-token-salt --name "My Token" --symbol "MTK" --decimals 18 --minter 0x456... --privateKey 0x... --accountAddress 0x...
+```
+
+### Register Custom Token
+
+```
+Usage: register-token [options]
+
+Register an existing token for cross-chain use
+
+Options:
+  --salt <salt>                      Salt for token registration
+  --tokenAddress <address>           Address of the existing token
+  --tokenManagerType <type>          Token manager type (native, mintBurnFrom,
+                                     lockUnlock, lockUnlockFee, mintBurn)
+  --operator <address>               Operator address (defaults to current
+                                     account)
+  -e, --env <env>                    environment (choices: "devnet-amplifier",
+                                     "mainnet", "stagenet", "testnet", default:
+                                     "testnet", env: ENV)
+  -y, --yes                          skip deployment prompt confirmation (env:
+                                     YES)
+  -p, --privateKey < privateKey >    private key for Starknet account(testnet
+                                     only, not required for offline tx
+                                     generation) (env: STARKNET_PRIVATE_KEY)
+  --accountAddress <accountAddress>  Starknet account address (env:
+                                     STARKNET_ACCOUNT_ADDRESS)
+  -h, --help                         display help for command
+```
+
+**Example:**
+```bash
+npx ts-node starknet/its/register-token.ts --salt my-salt --tokenAddress 0x... --tokenManagerType lockUnlock --privateKey 0x... --accountAddress 0x...
+```
+
+### Register Canonical Token
+
+```
+Usage: register-canonical-token [options]
+
+Register a canonical token for cross-chain use (uses lock/unlock mechanism)
+
+Options:
+  --tokenAddress <address>           Address of the canonical token to register
+  --factoryAddress <address>         InterchainTokenFactory address (defaults
```

Comment:
Why is this here but not in the other commands? :thinking:

Reply:
Factory address is passed for `deploy-remote-canonical-token`, `register-canonical-token`, `manage-service` set-factory-address action and `calculate-token-id`